facade/src/main/java/org/jboss/pnc/facade/providers/BuildConfigurationProviderImpl.java

```diff
@@ -350,6 +354,29 @@ public BuildConfigCreationResponse createWithScm(BuildConfigWithSCMRequest reque
         }
     }
 
+    @Override
+    public BuildConfiguration restoreRevision(Integer id, Integer rev) {
+        IdRev idRev = new IdRev(id, rev);
+        BuildConfigurationAudited buildConfigurationAudited = buildConfigurationAuditedRepository.queryById(idRev);
+        org.jboss.pnc.model.BuildConfiguration originalBC = repository.queryById(id);
+
+        if (buildConfigurationAudited == null || originalBC == null) {
+            throw new NotFoundException("BuildConfigurationAudited with [id=" + id + ", rev=" + rev + "] does not exists");
```

Comment:
This type of exception should be thrown in the endpoint implementation.

Reply:
OK. Changed.